Doc (#15)
enonic/app-google-analytics: docs/index.adoc

## Tracking

> Setting up the tracking only required one value to the application set on the site. Setting up the widget requires a couple addition steps.

Reviewer: ```suggestion
To turn on tracking script on your site, fill out the "Measurement Id" field and check off the "Enable tracking" checkbox in the app config in Content Studio. Setting up the widget requires a couple of additional steps.
```

> The app requires a google analytics acount setup for the site you want to track.

Reviewer: ```suggestion
The app requires a Google Analytics account setup for a site you want to track.
```

In google analytics go to admin settings

> The settings are located on the bottom left

Reviewer: ```suggestion
In Google Analytics go to Admin Settings located on the bottom left:
```

image::images/AdminButton.png[]

> There should be a datastream connected to the site for tracking

Reviewer: ```suggestion
There should be a data stream connected to the site you are setting up tracking for:
```

image::images/AdminPanel.png[]

> Opening up the data stream and select the one for the site we want to track.

Reviewer: ```suggestion
Open the data stream and select one for your site:
```

image::images/DataStream.png[]

> Find its measureId and copy the value. (Use the icon next to it)

Reviewer: ```suggestion
Locate "Measurement Id" and use the icon next to it to copy value to the clipboard.
```

> In content studio add the application to your site

Reviewer: ```suggestion
In Content Studio add the "Google Analytics" application to your site, then open app config (click the "pencil icon"):
```

And open up settings for this app (pencil icon)

image::images/SiteApplication.png[]

> In the site settings simply add a new measureId and enable tracking.

Reviewer: ```suggestion
Paste value from the clipboard to the "Measurement Id" config field and enable tracking by checking off "Enable tracking" checkbox.
```

image::images/AppSettings.png[]

> Thats it. If you publish the site now the tracking will be live and you can track user data.

Reviewer: ```suggestion
That's it. When you publish the site, the tracking script will go live and will start collecting data for analytics.
```

## Widget

Setting up the widget has a few more steps, but gives a nice overview over the statistics of the site.

> The next setup will require file access to the servers configuration files

Reviewer: ```suggestion
The following setup will require file access to configuration files located on the server where Enonic XP is hosted.
```

In the analytics admin page. (Cog wheel at the bottom left of the page)

> Find the Property Settings and press it to show its related information.

Reviewer: ```suggestion
In the Google Analytics admin page (cog wheel at the bottom left of the page) locate the Property Settings  and click to show related information:
```

image::images/PropertySettings.png[]

> When opening it find its property ID and copy it

Reviewer: ```suggestion
Locate Property ID under Property Settings and copy it to the clipboard:
```

image::images/PropertyId.png[]

> Paste it into the app settings in content studio

Reviewer: ```suggestion
Paste it into the app config in Content Studio:
```

image::images/AppSettingsPropertyId.png[]

> Now that we are done with the site setup. We just have to configuration couple of apis and setting the authentication

Reviewer: ```suggestion
Now that we are done with the site setup we just have to configure a couple of APIs and set up authentication.
```

This is done with a service account in the app configuration file.

First create an account in https://console.cloud.google.com
This is where we will create a service account for authentication with our app.

After creating an account open up the IAM & Admin page and the Service Accounts sub menu

image::images/CloudAdminServiceAccounts.png[]

Create a new service account. This will be used to access the analytics data.
Give it any name and generate an email address for the service account.

image::images/ServiceAccountCreate.png[]

After creating the serivice we need to enable the apis that it will access

Go to our APIs & Services on https://console.cloud.google.com/apis/dashboard

> On the page press the button to enable apis and services

Reviewer: ```suggestion
Go to "APIs & Services" on https://console.cloud.google.com/apis/dashboard and click the "+ Enable APIs and Services" button:
```

image::images/Enable_APIs.png[]

Enable the *Google Analytics Data API*

It should now have access to the api. We will later add it to our google analytics service.

With this service we will need to create a access key.
Click the new service key that we just created an go to the keys tab.

image::images/ServiceAccountKeys.png[]

> Add a new key. Its needs to be of type *json*

Reviewer: ```suggestion
Add a new key of type *json*.
```

Securley store this key. We will need to upload it to the server hosting enonic.

> Go back to the overview of all accounts and copy the email adresse of our service

Reviewer: ```suggestion
Go back to the overview of all accounts and copy the email address of our service.
```

We need to copy the data to our home/config folder on the server

The configuration files on the server are located here:

`/home/config`

We need to create a configuration file for our app.

`com.enonic.app.ga.cfg`

We need to put our developer key in the same folder.
This allows our app to access it.

now in the `com.enonic.app.ga.cfg` file we need to add these values:

ga.credentialPath = ${xp.home}/config/<key_file_name.json>

> Replce key_file_name with the filename of the key we downloaded.

Reviewer: ```suggestion
Replace key_file_name with the filename of the key we downloaded.
```

> Now the credentials are setup.

Reviewer: ```suggestion
Now the credentials are set up.
```
